Server Extension gets NPE if afterAll fails with a specific class without a suite

### Describe the bug

We have checks to print out that a suite failed during the afterAll, but if there is no suite and it is a specific class it will fail with a NPE when trying to fail the assertion.

```

java.lang.RuntimeException: java.lang.NullPointerException: Cannot invoke "java.lang.Class.getName()" because "this.suite" is null

	at org.infinispan.server.test.junit5.CustomInfinispanExtension.lambda$invokeMethodIfPresent$1(CustomInfinispanExtension.java:36)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1117)
	at org.infinispan.server.test.junit5.CustomInfinispanExtension.invokeMethodIfPresent(CustomInfinispanExtension.java:27)
	at org.infinispan.server.test.junit5.CustomInfinispanExtension.invokeMethodIfPresent(CustomInfinispanExtension.java:22)
	at org.infinispan.server.test.junit5.CustomInfinispanExtension.afterAll(CustomInfinispanExtension.java:43)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Class.getName()" because "this.suite" is null
	at org.infinispan.server.test.junit5.AbstractServerExtension.afterAll(AbstractServerExtension.java:147)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.infinispan.server.test.junit5.CustomInfinispanExtension.lambda$invokeMethodIfPresent$0(CustomInfinispanExtension.java:32)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at org.infinispan.server.test.junit5.CustomInfinispanExtension.lambda$invokeMethodIfPresent$1(CustomInfinispanExtension.java:32)
	... 6 more
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/server/testdriver/junit5/src/main/java/org/infinispan/server/test/junit5/AbstractServerExtension.java
+++ b/server/testdriver/junit5/src/main/java/org/infinispan/server/test/junit5/AbstractServerExtension.java
@@ -111,12 +111,16 @@ public abstract class AbstractServerExtension implements BeforeAllCallback, Afte
 
    @Override
    public final void beforeAll(ExtensionContext context) throws Exception {
+      Class<?> testClass = context.getRequiredTestClass();
       try {
          initSuiteClasses(context);
 
          // Inject all classes that are using a static @InfinispanServer(ClusteredIT.class) field.
-         Class<?> testClass = context.getRequiredTestClass();
-         log.infof("Starting test suite: %s for test %s", suite, testClass);
+         if (suite != null) {
+            log.infof("Starting test suite: %s for test %s", suite, testClass);
+         } else {
+            log.infof("Starting test %s", testClass);
+         }
 
          injectExtension(testClass, this);
 
@@ -128,7 +132,11 @@ public abstract class AbstractServerExtension implements BeforeAllCallback, Afte
          }
          onTestsStart(context);
       } catch (Throwable t) {
-         Assertions.fail(String.format("Failed during '%s#beforeAll' suite execution", suite.getName()), t);
+         if (suite != null) {
+            Assertions.fail(String.format("Failed during '%s#beforeAll' suite execution", suite.getName()), t);
+         } else {
+            Assertions.fail(String.format("Failed during '%s#afterAll' execution", testClass), t);
+         }
       }
    }
 
@@ -136,15 +144,24 @@ public abstract class AbstractServerExtension implements BeforeAllCallback, Afte
 
    @Override
    public final void afterAll(ExtensionContext context) {
+      String testName = String.valueOf(context.getTestClass().orElse(null));
       try {
-         log.infof("Finishing suite: %s for test %s", suite, context.getTestClass().orElse(null));
+         if (suite != null) {
+            log.infof("Finishing suite: %s for test %s", suite, testName);
+         } else {
+            log.infof("Finishing test %s", testName);
+         }
          cleanupSuiteClasses(context);
          // Only stop the extension resources when all tests in a Suite have been completed
          if (suiteTestClasses.isEmpty()) {
             onTestsComplete(context);
          }
       } catch (Throwable t) {
-         Assertions.fail(String.format("Failed during '%s#afterAll' suite execution", suite.getName()), t);
+         if (suite != null) {
+            Assertions.fail(String.format("Failed during '%s#afterAll' suite execution", suite.getName()), t);
+         } else {
+            Assertions.fail(String.format("Failed during '%s#afterAll' execution", testName), t);
+         }
       }
    }
 

